src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -4126,5 +4127,1053 @@ static void Main()
+                // (8,23): error CS1002: ; expected
+                //         scoped scoped ref R z = ref x;
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "ref").WithLocation(8, 23));
+        }
+
+        [Fact]
+        public void LocalScope_03()
+        {
+            var source =
+@"scoped scoped R x = default;
+ref scoped scoped R y = ref x;
+scoped scoped ref R z = ref x;
+ref struct R { }
+";
+            var comp = CreateCompilation(source);
+            // Duplicate scoped modifiers result are parse errors rather than binding errors.
+            comp.VerifyDiagnostics(
+                // (1,8): error CS1031: Type expected
+                // scoped scoped R x = default;
+                Diagnostic(ErrorCode.ERR_TypeExpected, "scoped").WithArguments("scoped").WithLocation(1, 8),
+                // (1,17): warning CS0219: The variable 'x' is assigned but its value is never used
+                // scoped scoped R x = default;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "x").WithArguments("x").WithLocation(1, 17),
+                // (2,12): error CS0118: 'scoped' is a variable but is used like a type
+                // ref scoped scoped R y = ref x;
+                Diagnostic(ErrorCode.ERR_BadSKknown, "scoped").WithArguments("scoped", "variable", "type").WithLocation(2, 12),
+                // (2,19): error CS8174: A declaration of a by-reference variable must have an initializer
+                // ref scoped scoped R y = ref x;
+                Diagnostic(ErrorCode.ERR_ByReferenceVariableMustBeInitialized, "R").WithLocation(2, 19),
+                // (2,19): warning CS0168: The variable 'R' is declared but never used
+                // ref scoped scoped R y = ref x;
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "R").WithArguments("R").WithLocation(2, 19),
+                // (2,21): error CS1003: Syntax error, ',' expected
+                // ref scoped scoped R y = ref x;
+                Diagnostic(ErrorCode.ERR_SyntaxError, "y").WithArguments(",", "").WithLocation(2, 21),
+                // (3,1): error CS0118: 'scoped' is a variable but is used like a type
+                // scoped scoped ref R z = ref x;
+                Diagnostic(ErrorCode.ERR_BadSKknown, "scoped").WithArguments("scoped", "variable", "type").WithLocation(3, 1),
+                // (3,8): warning CS0168: The variable 'scoped' is declared but never used
+                // scoped scoped ref R z = ref x;
+                Diagnostic(ErrorCode.WRN_UnreferencedVar, "scoped").WithArguments("scoped").WithLocation(3, 8),
+                // (3,15): error CS1003: Syntax error, ',' expected
+                // scoped scoped ref R z = ref x;
+                Diagnostic(ErrorCode.ERR_SyntaxError, "ref").WithArguments(",", "ref").WithLocation(3, 15));
+        }
+
+        [Fact]
+        public void LocalScope_04()
+        {
+            var source =
+@"scoped s1 = default;
+ref scoped s2 = ref s1;
```

Comment:
I didn't see tests for `ref readonly scoped` locals, but if such tests are found to be useful, they can wait for a subsequent PR.